crates/sync/anvil/src/tracker.rs

```diff
@@ -249,3 +277,42 @@ pub async fn fetch_erc20_metadata(address: Address, client: &Provider<Http>) ->
         decimals: contract.decimals().call().await.unwrap_or_default(),
     }
 }
+
+pub async fn fetch_erc721_data(
+    erc721_token: Erc721Token,
+    client: &Provider<Http>,
+) -> Erc721TokenData {
+    let contract = IERC721::new(erc721_token.contract, Arc::new(client));
+
+    let contract_uri = contract
+        .token_uri(erc721_token.token_id)
+        .call()
+        .await
+        .unwrap();
+
+    let mut md = "".to_string();
+    if contract_uri.contains("data:application/json;base64,") {
+        let byte_string = contract_uri
+            .strip_prefix("data:application/json;base64,")
+            .unwrap();
+
+        let decoded_uri = general_purpose::STANDARD.decode(byte_string);
+        md = String::from_utf8(decoded_uri.unwrap()).unwrap();
+    } else if contract_uri.contains("ipfs://") {
+        let contract_uri = contract_uri.replace("ipfs://", "https://ipfs.io/ipfs/");
+        let response = reqwest::get(contract_uri.clone()).await.unwrap();
+
+        if response.status().is_success() {
+            md = response.text().await.unwrap();
+        }
+    } else {
+        // TODO: handle exceptions
```

Comment:
If the tokenURI is nor base64 encoded, nor an ipfs URL, then it should be handled differently.